### DISCUSSION TITLE
Copilot is acting like a know-it-all lately

### DISCUSSION THREAD

**Discussion by Mrugalla** [community/community#60378] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I was always a huge fan of GIthub Copilot. Been using it since release to help me code VST plugins. But lately it's acting up. I mean, I liked when it sometimes just magically came up with a long piece of code that somehow works. But it's getting out of hand. Not only that loading times are just much longer on such long suggestions. But the results are also not usable. Sometimes I just want it to tell me what belongs into the line I'm currently in and it suggests to me the next 4 methods already as well and I'm just like "dude, I already wrote these methods, and you'd know that if I had copied them over this method already" but I don't wanna copy everything around all the time.

Ideally GIthub Copilot should just have a setting that tells it to more likely create output of a different length. Like not a 100% restriction on text length, but just a higher likelyness to produce less code in just one suggestion.

Or alternatively a hotkey that tells it to only take the first line of the current suggestion.

Or alternatively a way to select part of a suggestion to manifest it, instead of having to confirm to everything and then manually delete some of it.

**Comment by Mrugalla** [comment]
Srsly guys, the current implementation is garbage on so many levels. Please change it back or fix it in whatever way! You can't even let it write what Pi and Tau is without it suggestion a million additional methods. It even sometimes suggests methods that don't even fit to the file you're in. Like I was just working on some soft- and harclippers in my distortion header, then I went to a different file to work there and it kept on suggestion clipping functions. It remembered I was working on that, but couldn't understand that I went somewhere else. What is this? :D Who thought that was a good idea?

**Reply by max06** [reply]
Copilot uses your open files as context for suggestions. If you close everything you don't need it should stop doing that.

But I need to agree: The suggestions are getting worse. For me it's currently adding markdown codefences at the end of each suggestion...

**Reply by Mrugalla** [reply]
Yeah, totally. It makes me wonder how good my money would be spent if I actually spent money on Copilot. Because atm I just got it as a student, which I'm very grateful for, because I see that as an evaluation time. And the whole time I felt like this really adds something to my workflow and would be worth some cash once I'm not a student anymore, but now considering that it can just randomly become worse when the developers wanna try something new, that doesn't work out, makes me wonder how worth it would be. And I think it kinda depends on how they treat such happenings. Will I get a reply by someone who is at least a bit in charge of things? Maybe an explanation what's the reason for the problems, or a little road map on where we are heading with this tool would be cool to confirm that this is going to be fixed in some way

**Comment by Mrugalla** [comment]
Update: It seems to work fine again, but I'm not sure, because I also work on a different machine again. If this is what made the difference I'd still consider it some kind of bug, because I'd assume an AI tool, that doesn't process itself locally, to work the same on all machines ofc, especially considering that I was at a place with much better internet a week ago

**Reply by max06** [reply]
I'm having two days off due to the heatwave here (also called summer). I'll give it another shot tomorrow!
